Parent/Parent/Details/AssignmentDetailsViewController.swift

```diff
@@ -76,9 +91,60 @@ class AssignmentDetailsViewController: AssignmentDetailViewController {
         fatalError("init(coder:) has not been implemented")
     }
 
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        student.refresh()
+        teachers.refresh()
+        assignment.refresh()
+    }
+
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        let scheme = ColorScheme.observee(studentID)
-        navigationController?.navigationBar.useContextColor(scheme.color)
+        navigationController?.navigationBar.useContextColor(colorScheme.color)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        if replyButton == nil { configureComposeMessageButton() }
+    }
+
+    func configureComposeMessageButton() {
+        guard ExperimentalFeature.parentInbox.isEnabled else { return }
+        let buttonSize: CGFloat = 56
+        let margin: CGFloat = 16
+        let bottomMargin: CGFloat = 50
+
+        replyButton = FloatingButton(frame: CGRect(x: 0, y: 0, width: buttonSize, height: buttonSize))
+        replyButton?.setImage(UIImage.icon(.reply, .solid), for: .normal)
+        replyButton?.tintColor = .named(.white)
+        replyButton?.backgroundColor = colorScheme.color
+        if let replyButton = replyButton { view.superview?.addSubview(replyButton) }
+
+        let metrics: [String: CGFloat] = ["buttonSize": buttonSize, "margin": margin, "bottomMargin": bottomMargin]
+        replyButton?.addConstraintsWithVFL("H:[view(buttonSize)]-(margin)-|", metrics: metrics)
+        replyButton?.addConstraintsWithVFL("V:[view(buttonSize)]-(bottomMargin)-|", metrics: metrics)
+        replyButton?.addTarget(self, action: #selector(actionReplyButtonClicked(_:)), for: .primaryActionTriggered)
+    }
+
+    @IBAction func actionReplyButtonClicked(_ sender: UIButton) {
+        sender.isEnabled = false
+        replyStarted = true
+        messagingReady()
+    }
+
+    func messagingReady() {
+        let pending = teachers.pending || student.pending || assignment.pending
+        if !pending && replyStarted {
+            guard let a = assignment.first else { return }
+            let name = student.first?.fullName ?? ""
+            let template = NSLocalizedString("Regarding: %@, %@", comment: "Regarding <John Doe>, < assignment >")
+            let subject = String.localizedStringWithFormat(template, name, a.name)
+            let context = ContextModel(.course, id: courseID)
+            let body = a.htmlURL.absoluteString
+            let recipients = teachers.map { APIConversationRecipient(searchRecipient: $0) }
+            let r: Route = Route.compose(body: body, context: context, recipients: recipients, subject: subject)
```

Comment:
```suggestion
            let r: Route = Route.compose(body: nil, context: context, recipients: recipients, subject: subject, hiddenMessage: a.htmlURL.absoluteString)
```

Reply:
wow github, this little feature `suggested change` is cool